File: indicatif-ffi-py/examples/download.py
import time
from indicatif.progress import ProgressBar, ProgressStyle


def main():
    downloaded = 0
    total_size = 231231231

    ps = ProgressStyle()
    ps.set_progress_chars("#>-")
    ps.set_template("{spinner:.green} [{elapsed_precise}] "
                    "[{bar:40.cyan/blue}] {bytes}/{total_bytes} "
                    "({eta})")

    pb = ProgressBar(total_size)
    pb.set_style(ps)

    while downloaded < total_size:
        new = min(downloaded + 223211, total_size)
        downloaded = new

        pb.set_position(new)
        time.sleep(12 / 1000)

    pb.finish_with_message("downloaded")


if __name__ == '__main__':
    main()

File: indicatif-ffi-go-cgo/examples/download.go
package main

import (
	"time"

	"github.com/ARwMq9b6/indicatif-ffi/indicatif-ffi-go-cgo"
)

func main() {
	var downloaded uint64
	var total_size uint64 = 231231231

	style := indicatif.NewProgressStyleWithDefaultBar()
	style.SetTemplate("{spinner:.green} [{elapsed_precise}] [{bar:40.cyan/blue}] {bytes}/{total_bytes} ({eta})")
	style.SetProgressChars("#>-")

	pb := indicatif.NewProgressBar(total_size)
	pb.SetStyle(style)

	for downloaded < total_size {
		if downloaded += 223211; downloaded > total_size {
			downloaded = total_size
		}
		pb.SetPosition(downloaded)
		time.Sleep(12 * time.Millisecond)
	}

	pb.FinishWithMessage("downloaded")
}
